demdata-cms/docker-compose.dev.yml

```diff
@@ -0,0 +1,56 @@
+version: "3"
+services:
+  cms:
+    build: .
+    ports:
+      - 8000:8000
+    command: ["python3", "manage.py", "runserver", "0.0.0.0:8000"]
+    container_name: core_cms
+    hostname: core_cms
```

Comment:
**Idea.** Change all—not just these two lines—`core_cms` to `demdata_cms` to distinguish multiple CMS project development.

```suggestion
    container_name: demdata_cms
    hostname: demdata_cms
```

<details><summary>Use Case</summary>
<img width="977" alt="demdata-cms docker naming proposal" src="https://user-images.githubusercontent.com/62723358/198847201-6172d7bc-628f-4811-8b23-674b11bb0910.png">
</details>

Reply:
**Uh oh.** No longer just an idea. At least some of these must change to avoid this conflict error I always forget about when also running Core CMS, or another custom CMS, locally:

```
Error response from daemon: Conflict. The container name "/core_cms_postgres" is already in use by container "3138b5b0e2e5d377002afe73e5e01ccdec6f95e7d0cf746440d35d78476d3678". You have to remove (or rename) that container to be able to reuse that name.
```

Reply:
You're right. Good catch!

Reply:
Actually, I looked into this further and we should discuss Monday. Some of these names need to stay in order to be compatible with Core-Portal code. Particularly `core_cms_postgres`

Reply:
Okay. Thanks. Docker containers still trip me up.

<details><summary>Thoughts</summary>

Wes, you had thought `docker-compose.dev.yml` is only used locally, so how could it matter what Portal wants. But then you saw that the `docker-compose.___.override.yml` files do not set container names, which suggests the `core_cms_` prefix is necessary.

</details>

<details><summary>Alternative</summary>

Wes, perhaps, to allow multi-CMS local development for others, you just need to streamline and commit the `docker-compose.dev.unique.yml` you created for APCD and use locally. Or maybe you learn from Frank that there is an even better way.

```yaml
# To isolate APCD CMS development into its own database and volumes

# NOTE: The extends keyword is NOT in Compose version v3, but IS in latest spec
# SEE: https://docs.docker.com/compose/extends/#extending-services
# SEE: https://github.com/moby/moby/issues/31101#issuecomment-922378041
# version: "3"
services:
  cms:
    extends:
      file: ./docker-compose.dev.yml
      service: cms
    container_name: apcd_cms
    hostname: apcd_cms

  postgres:
    extends:
      file: ./docker-compose.dev.yml
      service: postgres
    hostname: apcd_cms_postgres
    container_name: apcd_cms_postgres
    volumes:
      - apcd_cms_postgres_data:/var/lib/postgresql/data

  elasticsearch:
    extends:
      file: ./docker-compose.dev.yml
      service: elasticsearch
    container_name: apcd_cms_elasticsearch
    volumes:
      - ./elasticsearch.yml:/usr/share/elasticsearch/config/elasticsearch.yml
      - apcd_cms_es_data:/usr/share/elasticsearch/data

volumes:
  apcd_cms_postgres_data:
  apcd_cms_es_data:

networks:
  core_cms_net:
```

</details>

Reply:
<details><summary>Note to Self (when I refer back to this lesson)</summary>

The [screenshot](https://user-images.githubusercontent.com/62723358/198847201-6172d7bc-628f-4811-8b23-674b11bb0910.png) neglects to show the containers—it just shows groups. The container group names are unique (and come from the name of the directory in which the `Dockerfile` lives.). yet, the container (not group) names still do clash and cause error **only if** multiple are running at once...

So just don't run multiple at once… In which case, do you still need `docker-compose.dev.unique.yml`?

</details>

Reply:
It turns out that you were right! I've made the changes and pushed them up!